tests/features/typing_pep563_test.py

```diff
@@ -0,0 +1,175 @@
+
+            'from __future__ import annotations\n'
+            'def foo(var: MyClass) -> MyClass:\n'
+            '   x: MyClass\n',
+
+            id='Simple annotation',
+        ),
+        pytest.param(
+            'from __future__ import annotations\n'
+            'def foo(\n'
+            '    var0: Tuple["MyClass"],\n'
+            '    *,\n'
+            '    var2: Tuple["MyClass"],\n'
+            ') -> Tuple["MyClass"]:\n'
+            '   x: Tuple["MyClass"]\n',
+
+            'from __future__ import annotations\n'
+            'def foo(\n'
+            '    var0: Tuple[MyClass],\n'
+            '    *,\n'
+            '    var2: Tuple[MyClass],\n'
+            ') -> Tuple[MyClass]:\n'
+            '   x: Tuple[MyClass]\n',
+
+            id='Tuple',
+        ),
+        pytest.param(
+            'from __future__ import annotations\n'
+            'def foo(\n'
+            '   var: Callable[["MyClass"], None]\n'
+            ') -> Callable[["MyClass"], None]:\n'
+            '   x: Callable[["MyClass"], None]\n',
+
+            'from __future__ import annotations\n'
+            'def foo(\n'
+            '   var: Callable[[MyClass], None]\n'
+            ') -> Callable[[MyClass], None]:\n'
+            '   x: Callable[[MyClass], None]\n',
+
+            id='List within Callable',
+        ),
+        pytest.param(
+            'from __future__ import annotations\n'
+            'class Foo(NamedTuple):\n'
+            '    x: "MyClass"\n',
+
+            'from __future__ import annotations\n'
+            'class Foo(NamedTuple):\n'
+            '    x: MyClass\n',
+
+            id='NamedTuple',
+        ),
+        pytest.param(
+            'from __future__ import annotations\n'
+            'x: List[List[List["MyClass"]]]\n',
+
+            'from __future__ import annotations\n'
+            'x: List[List[List[MyClass]]]\n',
+
+            id='Nested',
+        ),
+    ),
+)
+def test_fix_typing_pep563(s, expected):
+    ret = _fix_plugins(s, settings=Settings(min_version=(3, 7)))
+    assert ret == expected
+
+
+@pytest.mark.xfail(
+    sys.version_info < (3, 8),
+    reason='posonly args not available in Python3.7',
+)
+@pytest.mark.parametrize(
+    ('s', 'expected'),
+    (
+        pytest.param(
+            'from __future__ import annotations\n'
+            'def foo(\n'
+            '    var0: Tuple["MyClass"],\n'
+            '    /,\n'
+            '    var1: Tuple["MyClass"],\n'
+            '    *,\n'
+            '    var2: Tuple["MyClass"],\n'
+            ') -> Tuple["MyClass"]:\n'
+            '   x: Tuple["MyClass"]\n',
+
+            'from __future__ import annotations\n'
+            'def foo(\n'
+            '    var0: Tuple[MyClass],\n'
+            '    /,\n'
+            '    var1: Tuple[MyClass],\n'
+            '    *,\n'
+            '    var2: Tuple[MyClass],\n'
+            ') -> Tuple[MyClass]:\n'
+            '   x: Tuple[MyClass]\n',
+
+            id='Tuple',
+        ),
+    ),
+)
```

Comment:
probably not much of a need for `parametrize` here -- there's only one case.  also I'd stick to just testing what you intend to test here (cut down the argument list, also the `id` here is bogus so I'd just leave it out)